src/components/Undo/Undo.js

```diff
@@ -0,0 +1,59 @@
+const React = require("react");
+const {connect} = require("react-redux");
+const actions = require("actions/index");
+const gBlockedLinks = require("lib/blockedLinks");
+
+const Undo = React.createClass({
+  /**
+   * The undo dialog's timeout in miliseconds.
+   */
+  HIDE_TIMEOUT_MS: 15000,
+  timeout: null,
+
+  /**
+   * Hides the undo dialog.
+   */
+  hide() {
+    this.props.dispatch(actions.setUndoDialogVisibility(false));
+    clearTimeout(this.timeout);
```

Comment:
the timeout is being cleared, but your are not resetting this.timeout to 0 or null.